cloudsync/api.py

```diff
@@ -38,71 +41,102 @@
+
+
+def process_hls_outputs(outputs: list, video: Video):
+    """
+    Process HLS outputs and create VideoFile objects.
+    Args:
+        outputs (list): List of HLS output details.
+        video (Video): The video object to associate with the outputs.
+    """
+
+    # Process HLS playlists
+    for playlist in outputs:
+        playlist_key = ""
+        for file_path in playlist.get("outputFilePaths", []):
+            if file_path.endswith("__index.m3u8"):
+                playlist_key = file_path
+
                 VideoFile.objects.update_or_create(
-                    # This assumes MP4 encoding
-                    s3_object_key="{}".format(
-                        output["Key"].replace(RETRANSCODE_FOLDER, "")
-                    ),
+                    s3_object_key=playlist_key.replace(RETRANSCODE_FOLDER, ""),
                     defaults={
                         "video": video,
                         "bucket_name": settings.VIDEO_S3_TRANSCODE_BUCKET,
-                        "encoding": EncodingNames.DESKTOP_MP4,
-                        "preset_id": settings.ET_MP4_PRESET_ID,
+                        "encoding": EncodingNames.HLS,
+                        "preset_id": "",
+                    },
+                )
+
+
+def process_mp4_outputs(outputs: list, video: Video):
+    """
+    Process MP4 outputs and create VideoFile objects.
+    Args:
+        outputs (list): List of MP4 output details.
+        video (Video): The video object to associate with the outputs.
+    """
+
+    # Process MP4 outputs
+    for playlist in outputs:
+        playlist_key = ""
+        for file_path in playlist.get("outputFilePaths", []):
+            if file_path.endswith(".mp4"):
+                playlist_key = file_path
+
+                VideoFile.objects.update_or_create(
+                    s3_object_key=playlist_key.replace(RETRANSCODE_FOLDER, ""),
+                    defaults={
+                        "video": video,
+                        "bucket_name": settings.VIDEO_S3_TRANSCODE_BUCKET,
+                        "encoding": EncodingNames.MP4,
+                        "preset_id": "",
                     },
                 )
-            continue
-        thumbnail_pattern = output["ThumbnailPattern"].replace("{count}", "")
-        preset = get_et_preset(output["PresetId"])
-        bucket = get_bucket(settings.VIDEO_S3_THUMBNAIL_BUCKET)
-        for thumb in bucket.objects.filter(Prefix=thumbnail_pattern):
-            VideoThumbnail.objects.update_or_create(
-                s3_object_key=thumb.key.replace(RETRANSCODE_FOLDER, ""),
-                defaults={
-                    "video": video,
-                    "bucket_name": settings.VIDEO_S3_THUMBNAIL_BUCKET,
-                    "preset_id": output["PresetId"],
-                    "max_height": int(preset["Thumbnails"]["MaxHeight"]),
-                    "max_width": int(preset["Thumbnails"]["MaxWidth"]),
-                },
-            )
```

Comment:
Can you make sure we are not missing thumbnails ?
are we saving it somewhere else ?